cmd/local/local.go

```diff
@@ -0,0 +1,483 @@
+package local
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/kubefirst/kubefirst/configs"
+	"github.com/kubefirst/kubefirst/internal/argocd"
+	"github.com/kubefirst/kubefirst/internal/domain"
+	"github.com/kubefirst/kubefirst/internal/gitClient"
+	"github.com/kubefirst/kubefirst/internal/githubWrapper"
+	"github.com/kubefirst/kubefirst/internal/handlers"
+	"github.com/kubefirst/kubefirst/internal/helm"
+	"github.com/kubefirst/kubefirst/internal/k3d"
+	"github.com/kubefirst/kubefirst/internal/k8s"
+	"github.com/kubefirst/kubefirst/internal/metaphor"
+	"github.com/kubefirst/kubefirst/internal/progressPrinter"
+	"github.com/kubefirst/kubefirst/internal/services"
+	"github.com/kubefirst/kubefirst/internal/terraform"
+	"github.com/kubefirst/kubefirst/internal/vault"
+	"github.com/kubefirst/kubefirst/pkg"
+	"github.com/segmentio/analytics-go"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"log"
+	"os/exec"
+	"sync"
+	"syscall"
+	"time"
+)
+
+var (
+	useTelemetry   bool
+	dryRun         bool
+	silentMode     bool
+	gitOpsBranch   string
+	gitOpsRepo     string
+	awsHostedZone  string
+	metaphorBranch string
+	adminEmail     string
+	enableConsole  bool
+)
+
+func NewCommand() *cobra.Command {
+
+	localCmd := &cobra.Command{
+		Use:      "local",
+		Short:    "Kubefirst localhost installation",
+		Long:     "Kubefirst localhost enable a localhost installation without the requirement of a cloud provider.",
+		PreRunE:  validateLocal,
+		RunE:     runLocal,
+		PostRunE: runPostLocal,
+	}
+
+	localCmd.Flags().BoolVar(&useTelemetry, "use-telemetry", true, "installer will not send telemetry about this installation")
+	localCmd.Flags().BoolVar(&dryRun, "dry-run", false, "set to dry-run mode, no changes done on cloud provider selected")
+	localCmd.Flags().BoolVar(&silentMode, "silent", false, "enable silentMode mode will make the UI return less content to the screen")
+	// todo: get it from GH token , use it for console
+	localCmd.Flags().StringVar(&adminEmail, "admin-email", "", "the email address for the administrator as well as for lets-encrypt certificate emails")
+
+	localCmd.Flags().StringVar(&metaphorBranch, "metaphor-branch", "main", "metaphro application branch")
+	localCmd.Flags().StringVar(&gitOpsBranch, "gitops-branch", "main", "version/branch used on git clone - former: version-gitops flag")
+	localCmd.Flags().StringVar(&gitOpsRepo, "gitops-repo", "gitops", "")
+
+	localCmd.Flags().BoolVar(&enableConsole, "enable-console", true, "If hand-off screen will be presented on a browser UI")
```

Comment:
this will disable the support of the config file, right?

Reply:
Or do we plan to mix global vars with the current read config logic?

Reply:
I'll tackle that after merging